source/_posts/2017-05-12-openwrt_on_apu3.markdown

```diff
@@ -0,0 +1,500 @@
+platform in some generic use-cases. I'm trying to explain how to work with
+the device and use it in a generic manner. There is a part about the coreboot
+firmware, which could be used as a reference of how to start customizing it
+for own purposes.
+
+## Configuring the hardware
+
+At first, let's figure out some basic requirements for our new device:
+
+1. It will be wireless router with some advenced functionality provided by
+    OpenWRT.
+2. In order for it to be wireless, we need to add WiFi network adapters.
+3. I want it to be dual-band simultaneous connection, so we will need 2 separate
+    WiFi adapters.
+4. Operating system will be placed on µSD card.
+5. There will be an additional storage in the form of mSata disk.
+
+APU3 has 3 mPcie slots. Unfortunately it supports PCI express only on slot
+`mPCIe 1`, so WiFi card has to use it. For the second WiFi card, we could use
+`mPCIe 2` slot, but we would need USB only type, which are rare. Instead I'm
+using some cheap Ralink RT5370 based USB dongle WiFi adapter.
+`mPCIe 3` slot will be used for mSata drive.
+
+<a class="fancybox" rel="group" href="/assets/images/apu3_overall.jpg"><img src="/assets/images/apu3_overall.jpg" width=800 alt="" /></a>
+
+For the OS drive, I'll use some generic µSD card with adapter.
+
+`mPCIe 2` slot could be used in future for GSM modem or some other kind of
+USB device in the form of mPcie card.
+
+## Getting the sources
+
+We will use latest stable version which is Chaos Calmer, in order to be
+compatible with upstream packages. Thanks to that we can just use the `opkg`
+to download new version of packages from the main OpenWRT's repositories.
+
+The sources we need are located on [github](https://github.com/openwrt/openwrt).
+
+Let's clone the needed version:
+
+```sh
+$ git clone -b chaos_calmer https://github.com/openwrt/openwrt.git
+Cloning into 'openwrt'...
+remote: Counting objects: 360802, done.
+remote: Compressing objects: 100% (5/5), done.
+remote: Total 360802 (delta 4), reused 2 (delta 2), pack-reused 360795
+Receiving objects: 100% (360802/360802), 132.94 MiB | 8.24 MiB/s, done.
+Resolving deltas: 100% (241401/241401), done.
+```
+
+## Building
+
+To build our first image we first need to configure the OpenWRT:
+
+```sh
+$ cd openwrt
+$ make menuconfig
+```
+
+Our target is APU system, which has AMD x86_64 CPU. So let's use generic
+settings:
+
+* Target System > x86
+* Subtarget > x86_64
+
+... and then `Exit` and `make`.
+
+After compilation our image is in `bin/x86` dir. We need a SD card to burn
+the image and boot the system on the target platform. On my host system, card
+is present under the device file `/dev/sde`.
+
+> Warning! Carefully check the device the card is present on your system. This
+> is potentially dangerous operation and can lead to lost data, when used
+> wrong device!
+
+```sh
+cd bin/x86
+sudo dd if=openwrt-x86-64-combined-ext4.img of=/dev/sde bs=4M
+```
+
+## First boot
```

Comment:
@mek-x we should add information about login/pass - not all people are familiar with OpenWRT defaults.